pennylane/circuit_graph.py

```diff
@@ -12,305 +12,283 @@
 
     Args:
-        queue (list[Operation]): the quantum operations to apply
-        observables (list[Observable]): the quantum observables to measure
-        parameters (dict[int->list[(int, int)]]): Specifies the free parameters
-            of the quantum circuit. The dictionary key is the parameter index.
-            The first element of the value tuple is the operation index,
+        ops (Iterable[Operation]): quantum operations constituting the circuit, in temporal order
+        variable_deps (dict[int, list[ParDep]]): Free parameters of the quantum circuit.
+            The dictionary key is the parameter index.
+            The first element of the ParDep namedtuple is the operation index,
             the second the index of the parameter within the operation.
     """
-
-    def __init__(self, queue, observables, parameters=None):
-        self._queue = queue
-        self._observables = observables
-        self.parameters = parameters or {}
+    def __init__(self, ops, variable_deps):
+        self.variable_deps = variable_deps
 
         self._grid = {}
-        """dict[int, list[Command]]: dictionary representing the quantum circuit
-        as a grid. Here, the key is the wire number, and the value is a list
-        containing the Command object containing the operation/observable itself.
+        """dict[int, list[Operation]]: dictionary representing the quantum circuit as a grid.
+        Here, the key is the wire number, and the value is a list containing the operations on that wire.
         """
-
-        for idx, op in enumerate(queue + observables):
-            cmd = Command(
-                name=op.name, op=op, return_type=getattr(op, "return_type", None), idx=idx
-            )
-
+        for k, op in enumerate(ops):
+            op.queue_idx = k  # store the queue index in the Operation
             for w in set(op.wires):
-                if w not in self._grid:
-                    # wire is not already in the grid;
-                    # add the corresponding wire to the grid
-                    self._grid[w] = []
-
-                # Add the operation to the grid, to the end of the specified wire
-                self._grid[w].append(cmd)
-
-        self._graph = nx.DiGraph()
+                # Add op to the grid, to the end of wire w
+                self._grid.setdefault(w, []).append(op)
 
+        self._graph = nx.DiGraph()  #: nx.DiGraph: DAG representation of the quantum circuit
         # Iterate over each wire in the grid
-        for cmds in self._grid.values():
-            if cmds:
+        for wire in self._grid.values():
+            if wire:
                 # Add the first operation on the wire to the graph
                 # This operation does not depend on any others
-                self._graph.add_node(cmds[0].idx, **cmds[0]._asdict())
+                self._graph.add_node(wire[0])
 
-            for i in range(1, len(cmds)):
+            for i in range(1, len(wire)):
```

Comment:
This code block could be indented, as it is never executed if the above if statement is false.

Reply:
I removed the if statement, since empty wires should never end up in _grid.